auth_server/server/server.go

```diff
@@ -212,16 +213,29 @@ func (as *AuthServer) ParseRequest(req *http.Request) (*authRequest, error) {
 		for _, scopeStr := range req.Form["scope"] {
 			parts := strings.Split(scopeStr, ":")
 			var scope authScope
+
+			scopeType := parts[0]
+			var scopeClass string
+			if strings.HasPrefix(scopeType, "repository") {
```

Comment:
instead of prefix-suffix matching, can't we just compare to "repository(plugin)"?

Reply:
According to [this document](https://docs.docker.com/registry/spec/auth/scope/) there's a third scope that could be sent. This is why I went with matching tring parts.

Reply:
cool, that's a good document.
the way i read it, scope consists of resource type and optional class, i.e. `TYPE` or `TYPE(CLASS)`.
can you write a helper function that parses this format and returns resourceType, resourceClass and error (if rformat is not recognized) ?

Reply:
yes, let's do that!

Reply:
Ok, this is done! Thanks for the review :)